lib/core/versions/latest/Did.ts

```diff
@@ -1,10 +1,89 @@
 import Encoder from './Encoder';
+import ErrorCode from './ErrorCode';
 import Multihash from './Multihash';
+import { SidetreeError } from '../../Error';
 
 /**
- * Class containing reusable DID related operations.
+ * Class containing reusable Sidetree DID related operations.
  */
 export default class Did {
+
+  /** `true` if DID is short form; `false` if DID is long-form. */
+  public isShortForm: boolean;
+  /** DID method name. */
+  public didMethodName: string;
+  /** DID unique suffix. */
+  public uniqueSuffix: string;
+  /** Encoded DID Document if given DID is long-form, `undefined` otherwise. */
+  public encodedDidDocument?: string;
+  /** The short form. */
+  public shortForm: string;
+
+  /**
+   * Parses the input string as Sidetree DID.
+   * @param did Short or long-form DID string.
+   * @param didMethodName The expected DID method given in the DID string. The method throws SidetreeError if mismatch.
+   */
+  private constructor (did: string, didMethodName: string) {
+    if (!did.startsWith(didMethodName)) {
+      throw new SidetreeError(ErrorCode.DidIncorrectPrefix);
+    }
+
+    this.didMethodName = didMethodName;
+
+    const indexOfSemiColonChar = did.indexOf(';');
+    // If there is no semicolon, then DID can only be in short-form.
+    if (indexOfSemiColonChar < 0) {
+      this.isShortForm = true;
+    } else {
+      this.isShortForm = false;
+    }
+
+    if (this.isShortForm) {
+      this.uniqueSuffix = did.substring(didMethodName.length);
+    } else {
+      // This is long-form.
+      this.uniqueSuffix = did.substring(didMethodName.length, indexOfSemiColonChar);
+    }
+
+    if (this.uniqueSuffix.length === 0) {
+      throw new SidetreeError(ErrorCode.DidNoUniqueSuffix);
+    }
+
+    this.shortForm = didMethodName + this.uniqueSuffix;
+
+    // Get the encoded document if it's long-form.
+    if (!this.isShortForm) {
+      const didParameterString = did.substring(indexOfSemiColonChar + 1);
+
+      if (!didParameterString.startsWith('initial-values=')) {
```

Comment:
Why is this a `startsWith` check? There's no prohibition on other matrix parameters being included, and they are not ordered such that `initial-values` will always be first.

Reply:
What you are asking is supporting other DID parameters, which is currently not supported, thus only `initial-values` is allowed. If feature of supporting other DID parameters is a requirement and mainnet blocker, please file a issue.